docs/developer/architecture.md

```diff
@@ -1,6 +1,49 @@
 # Architecture
 
-___TO BE ADDED___
+**IMPORTANT:** It reflects only the structure as of v2.0.0 and may not reflect the structure of
+previous or future versions.
+
+Kubernetes Dashboard project consists of two main components. They are called here the
+frontend and the backend.
+
+The frontend is a single page web application that runs in a browser. It fetches all its
+business data from the backend using standard HTTP methods. It does not implement business logic,
+it only presents fetched data and sends requests to the backend for actions.
+
+The backend runs in a Kubernetes cluster as a Kubernetes service. Alternatively, it may run anywhere
+outside of the cluster, given that it can connect to the master. The backend is an HTTP server that
+proxies data requests to appropriate remote backends (e.g., Kubernetes API server or Heapster) or implements
+business logic. The backend implements business logic when remote backends APIs do not
+support required use case directly, e.g., “get a list of pods with their CPU usage metric
+timeline”. The figure below outlines the architecture of the project:
+
+![Architecture Overview](../images/architecture.png)
```

Comment:
ditto

Reply:
Heapster is still there, I was aware of this. We can update anyways. Will do it next week.